### PR TITLE
Add add-recommendation-1 to master

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently, when refreshing a page, coaches do not stay on the page they were.
- Currently, there is no tab calling for direct action of the potential customer.
- Currently, the fetching of cat and zombie data is not clean. It happens within the `render()` function.
- Currently, there are mostly useless quick actions for the coach.
- Currently, there is no way to get an appointment from the customer page.

**What is the new behavior (if this is a feature change)?**
- When refreshing a page, the coach stays on the page.
- A new tab calling for direct action has been added to the customer page.
- The way of fetching cat, zombie and goodboy data has been improved by using `componentDidMount` and `componentWillReceiveProps`. No more fetching is happening within the ReactJS `render()` function.
- Some more usefull links have been added to the coach quick actions. (Firebase, MS Office, SevDesk)
- A way to get appointments from the customer page has been added to the quick actions and the new tab.

**Other information**:
- The version has been updated to v1.4.0.

### CODE REVIEW

**Review by Aichnerc — COMMENTED** [review]
Looks good to me. I am certain the bois will find something though.

File: src/store/reducers/authReducer.js [hunk 1]
```diff
@@ -55,17 +44,19 @@ const authReducer = (state = initState, action) => {
...
       };
-    case "GETCATS_ERROR":
+    case "GETUSERS_ERROR":
       return {
         ...state,
-        cats: null,
+        users: null,
       };
     case "SIGNOUT_SUCCESS":
+      localStorage.removeItem("activePage");
```
Comment (schettn) [inline src/store/reducers/authReducer.js:59]: add newline

File: src/store/actions/authActions.js [hunk 2]
```diff
@@ -100,14 +100,44 @@ export const getZombies = () => {
...
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+
+    firestore
+      .collection("users")
+      .where("mode", "==", "goodboy")
+      .get()
+      .then((querySnapshot) => {
+        let goodboys = querySnapshot.docs.map((doc) => {
+          let data = doc.data();
```
Comment (schettn) [inline src/store/actions/authActions.js:128]: add newline

File: src/store/actions/authActions.js [hunk 3]
```diff
@@ -100,14 +100,44 @@ export const getZombies = () => {
...
+          data.uid = doc.id;
```
Comment (schettn) [inline src/store/actions/authActions.js:130]: add newline

File: src/components/organisms/CatList/index.jsx [hunk 4]
```diff
@@ -78,22 +78,26 @@ class CatList extends React.Component {
...
       this.setState({ sync: false });
     }
   }
 
-  toggle = (cat) => {
+  toggle = (user) => {
     if (!this.state.modal) {
       this.setState({
         modal: true,
         modalCat: {
```
Comment (schettn) [inline src/components/organisms/CatList/index.jsx:99]: Does it make sense to use all methods in connection with cats? 🐈
Or does cats use this website? I could be true.

File: src/components/pages/ProfilePage/index.jsx [hunk 5]
```diff
@@ -58,6 +59,13 @@ class ProfilePage extends React.Component {
   componentDidMount = () => {
     // Load welcoming picture
     this.getGreetingImg();
+    // Check if there is an active page
+    const activePage = localStorage.getItem("activePage");
```
Comment (schettn) [inline src/components/pages/ProfilePage/index.jsx:64]: add newline

File: src/components/pages/ProfilePage/index.jsx [hunk 6]
```diff
@@ -58,6 +59,13 @@ class ProfilePage extends React.Component {
   componentDidMount = () => {
     // Load welcoming picture
     this.getGreetingImg();
```
Comment (schettn) [inline src/components/pages/ProfilePage/index.jsx:61]: add newline

**Review by schettn — CHANGES_REQUESTED** [review]
Almost

**Review by LucaAllmaier — APPROVED** [review]
Looks good to me.

File: src/store/reducers/authReducer.js (hunk 1 above)
Comment (Aichnerc) [reply src/store/reducers/authReducer.js:59]: Nah. Looks great this way.

File: src/components/organisms/CatList/index.jsx (hunk 4 above)
Comment (Aichnerc) [reply src/components/organisms/CatList/index.jsx:99]: Cats are an entity in our customer acquisition workflow.

Comment (schettn) [reply src/components/organisms/CatList/index.jsx:99]: Cute

**Review by schettn — APPROVED** [review]
LGTM! 🌵

**Review by sweggerat — APPROVED** [review]
looks good

**Pull request merged**